Source/views/MacawView.swift

```diff
@@ -11,31 +11,178 @@ import AppKit
 /// You could create your own view extended from MacawView with predefined scene.
 ///
 
-internal class TouchInterceptionView: MView {
-    weak var macawView: MacawView?
+open class MacawView: MView {
+
+    internal var drawingView = DrawingView()
+
+    public let zoom = MacawZoom()
+
+    open var node: Node {
+        get { return drawingView.node }
+        set { drawingView.node = newValue }
+    }
+
+    open var contentLayout: ContentLayout {
+        get { return drawingView.contentLayout }
+        set { drawingView.contentLayout = newValue }
+    }
+
+    open override var contentMode: MViewContentMode {
+        get { return drawingView.contentMode }
+        set { drawingView.contentMode = newValue }
+    }
+
+    open var place: Transform {
+        get { return drawingView.place }
+    }
+
+    open var placeVar: Variable<Transform> {
+        get { return drawingView.placeVar }
+    }
+
+    open override var frame: CGRect {
+        get { return drawingView.frame }
+        set { drawingView.frame = newValue }
+    }
+
+    override open var intrinsicContentSize: CGSize {
+        get { return drawingView.intrinsicContentSize }
+    }
+
+    #if os(OSX)
+    open override var layer: CALayer? {
+        didSet {
+            guard self.layer != nil else {
+                return
+            }
+            initializeView()
+
+            drawingView.renderer = RenderUtils.createNodeRenderer(node, view: self)
+        }
+    }
+    #endif
+
+    @objc public convenience required init?(coder aDecoder: NSCoder) {
+        self.init(node: Group(), coder: aDecoder)
+    }
+
+    @objc public init?(node: Node, coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+
+        if let drawingView = DrawingView(node: node, coder: aDecoder) {
+            self.drawingView = drawingView
+        }
+
+        zoom.initialize(view: self, onChange: onZoomChange)
+    }
+
+    public convenience init(node: Node, frame: CGRect) {
+        self.init(frame: frame)
+
+        self.drawingView = DrawingView(node: node, frame: frame)
+    }
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        self.drawingView = DrawingView(frame: frame)
+
+        zoom.initialize(view: self, onChange: onZoomChange)
+    }
+
+    private func onZoomChange(t: Transform) {
+        if let viewLayer = drawingView.mLayer {
+            viewLayer.transform = CATransform3DMakeAffineTransform(t.toCG())
+        }
+    }
+
+    open override func didMoveToWindow() {
+        super.didMoveToWindow()
+
+        initializeView()
+    }
+
+    func initializeView() {
+
+        if !self.subviews.contains(drawingView) {
+            self.backgroundColor = .white
+            self.clipsToBounds = true
+            self.addSubview(drawingView)
+            drawingView.backgroundColor = .white
```

Comment:
I think drawingView should be transparent. If you keep it white, you will be unable to change its color.